oem/ibm/pfw-sms-utils/pfw_sms_menu.cpp

```diff
@@ -55,26 +56,27 @@ static int pamConversationFunction(int num_msg,
     }
 
     struct pfw_sms_pam_appdata* pfw_sms_appdata_ptr =
-        (struct pfw_sms_pam_appdata*)appdata_ptr;
-    *response_ptr = (pam_response*)calloc(
-        num_msg, sizeof(struct pam_response)); // PAM frees
-    if (*response_ptr == NULL)
+        reinterpret_cast<struct pfw_sms_pam_appdata *>(appdata_ptr);
+    // Malloc array of responses that PAM will free; required to use malloc.
+    *response_ptr = reinterpret_cast<struct pam_response*>(calloc(
+        num_msg, sizeof(struct pam_response))); // PAM will free
+    if (*response_ptr == nullptr)
     {
         return PAM_CONV_ERR;
     }
     const struct pam_message* msg_ptr = *message_ptr;
-    for (int msg_index = 0; msg_index < num_msg; ++msg_index)
+    for (const int msg_index : std::ranges::iota_view{1, num_msg})
```

Comment:
This change introduced a bug which is fixed by PR 523.  The range should start from zero.